Dockerfile

```diff
@@ -17,41 +17,44 @@ RUN curl -L https://github.com/wkhtmltopdf/packaging/releases/download/0.12.6-1/
     dpkg -i wkhtmltopdf.deb && \
     rm -f wkhtmltopdf.deb
 
-## Create a user for the web app.
-RUN \
-  addgroup --gid 9999 app && \
-  adduser --uid 9999 --gid 9999 --disabled-password --gecos "Application" app && \
-  usermod -L app
+# Create a user for the web app.
+RUN addgroup --gid 9999 app && \
+    adduser --uid 9999 --gid 9999 --disabled-password --gecos "Application" app && \
+    usermod -L app
+
+# Application directory
+RUN mkdir /app
+WORKDIR /app
 
+# Configuration
 ARG gemfile=Gemfile
 ENV BUNDLE_GEMFILE=${gemfile}
 ENV POIROT_STDOUT true
 ENV POIROT_SUPPRESS_RAILS_LOG true
-ENV PUMA_OPTIONS "--preload -w 4"
+ENV PUMA_OPTIONS "--preload -w 4 -p 3000"
 ENV NNDD_VERSION "cdx-0.11-pre7"
+ENV RAILS_ENV=production
 
 # Install gem bundle
 COPY Gemfile* cdx.gemspec cdx-api-elasticsearch.gemspec /app/
 COPY deps/ /app/deps/
-
 RUN bundle install --jobs 8 --deployment --without development test
 
 # Install the application
 ADD . /app
 
 # Precompile assets
-RUN bundle exec rake assets:precompile RAILS_ENV=production
+RUN bundle exec rake assets:precompile RAILS_ENV=${RAILS_ENV}
 
 # Download NNDD
-RUN \
-  mkdir -p /app/public/ && \
-  curl -L https://github.com/instedd/notifiable-diseases/releases/download/$NNDD_VERSION/nndd.tar.gz | tar -xzv -C /app/public/
+RUN mkdir -p /app/public/ && \
+    curl -L https://github.com/instedd/notifiable-diseases/releases/download/$NNDD_VERSION/nndd.tar.gz | tar -xzv -C /app/public/
 
 # Configure NNDD
 RUN /app/docker/config-nndd
 
 # Set permissions for tmp and log directories
 RUN mkdir -p /app/tmp /app/log && chown -R app:app /app/tmp /app/log
 
-# Add config files
-ADD docker/web-run /etc/service/web/run
+EXPOSE 3000
+CMD ["/app/docker/web-run"]
```

Comment:
The command actually starts `whenever` and `puma` processes. We might want to have a distinct container for `whenever`, like we do for `sidekiq`.

Alternative: use `sidekiq-cron` or `sidekiq-scheduler` instead.